co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/V2xTimeMessage.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2023 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.eclipse.mosaic.fed.infrastructure.ambassador;
+
+import java.net.InetAddress;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Message to be sent or received by the Infrastructure Device NS-3 Adapater interface
+ * NOTE: TODO See .ambassador for reference
+ 
+ */
+public class V2xTimeMessage {
```

Comment:
Why are we calling this a V2X Time Message. Isn't it just a V2X Message.

Reply:
I thought we needed 3 different messages from diagram in the PR description:
- `V2xRegistrationMessage` to communicate with `Registration Adapter`
- `V2xTimeMessage` to communicate with `Time Adapter`
- `InfrastructureV2xMessage` (which I did not define because there was `InfrastructureV2xMessageReception` that already existed, so I assumed the message exists0 to communicate with `Messages Adapter`

Please let me know if I am thinking it correctly

Reply:
You are correct in the message we need, but the time message does not need all this information. It only needs the current time and the time step number (see https://usdot-carma.atlassian.net/wiki/spaces/CRMSIM/pages/2475065345/XIL+MOSAIC+Time+Export+Ambassador+Detailed+Design). The message fields you have here are for InfrastructureV2XMessages. I think actually infrastructure and vehicle V2X Messages have the same format. @kjrush Is there somewhere we can put the V2XMessage Definition so it can be shared by CARMA Ambassador and Infrastructure Ambassador?

Reply:
Gotcha. I can delete all the contents, and the person who is implementing them can populate it.

Reply:
Also we should probably stick to one naming convention. Lets call the class InfrastructureRegistrationMessage and InfrastructureTimeMessage